packages/bundler/src/Bundler.ts

```diff
@@ -156,14 +156,18 @@ export class Bundler implements IBundler {
                   if (confirmations >= confirmedBlocks) {
                     clearInterval(intervalId);
                     resolve(userOpResponse);
+                    return;
                   }
+                } else {
+                  clearInterval(intervalId);
+                  resolve(userOpResponse);
+                  return;
                 }
-                clearInterval(intervalId);
```

Comment:
this is practically already in else

Reply:
Issue #374 arises precisely because this is not in an else block: if the `confirmations >= confirmedBlocks` condition is false, it should not `resolve`.

Reply:
Absolutely, that's an oversight on my end. I've changed it now.